numpy/lib/tests/test_nanfunctions.py

```diff
@@ -723,22 +723,22 @@ def test_ddof(self):
                 res = nf(_ndat, axis=1, ddof=ddof)
                 assert_almost_equal(res, tgt)
 
-    def test_ddof_too_big(self, recwarn):
+    def test_ddof_too_big(self):
         nanfuncs = [np.nanvar, np.nanstd]
         stdfuncs = [np.var, np.std]
         dsize = [len(d) for d in _rdat]
         for nf, rf in zip(nanfuncs, stdfuncs):
             for ddof in range(5):
-                with warnings.catch_warnings():
+                with warnings.catch_warnings(record=True) as w:
```

Comment:
FWIW, it looks like the concurrency safety guarantee of `catch_warnings` is only strong at CPython >= 3.14 per https://docs.python.org/3/library/warnings.html#concurrent-safety-of-context-managers.

I suspect that isn't too surprising though, with 3.14 generally considered better for free-threading support.

Reply:
@bwhitt7 has a followup to set up a pytest-run-parallel CI job by repurposing the 3.14t MacOS CI job.